tests/autologging/test_autologging_behaviors_unit.py

```diff
@@ -312,3 +314,124 @@ def test_autolog(disable=False, silent=False):
     # Verify that nothing is printed to the stderr-backed MLflow event logger, which would indicate
     # rerouting of warning content
     assert not stream.getvalue()
+
+
+def test_autolog_function_thread_safety(patch_destination):
```

Comment:
@WeichenXu123 Do you have any realistic example that doesn't work but would work with the fix in this PR?

Reply:
I think this is more of a bug fix.

User might enable / disable autolog in threads, this is not a typical use-case, but user might do it, and it doesn't raise error, BUT, this might cause the autolog patching can't be reverted correctly when autologging disabled, and might cause bug hard to debug when the code continues running.